packages-exp/auth-compat-exp/src/auth.ts

```diff
@@ -54,12 +49,18 @@ export class Auth
 
     // Note this is slightly different behavior: in this case, the stored
     // persistence is checked *first* rather than last. This is because we want
-    // the fallback (if no user is found) to be the stored persistence type
-    const storedPersistence = _getPersistenceFromRedirect(this.auth);
-    const persistences = storedPersistence ? [storedPersistence] : [];
-    persistences.push(exp.indexedDBLocalPersistence);
+    // to prefer stored persistence type in the hierarchy.
```

Comment:
@samhorlbeck please audit the old and new code closely since I'm not sure if I fully understood the logic.

Reply:
This makes sense to me